### DISCUSSION TITLE
Request for getInvoice extension: isFullyPaid and amountPaid

### DISCUSSION THREAD

**Discussion by mp21-org** [btcpayserver/btcpayserver#3110] [Technical Support]
As discussed in [this issue](https://github.com/btcpayserver/btcpayserver-greenfield-php/issues/35), we ( @petzsch and @woutersamaey and @ndeet ) would like to propose an extension to the getInvoice API response:

As a webshop owner who keeps prepaid accounts for all of his customers, I would like to accuratly account partialy and overpaid Invoices. Because an invoice can be marked as Settled even though it was underPaid, I can't determine (without doing heavy math on the paymentmethods response) if the invoice was in fact fully paid and what amount in fiat currency was actually paid.

Additionally the question if an invoice is to be considered fullyPaid depends potentially on the paymentTolerance set in the invoice and if the user paid within this tollerance ... allowing to configure this behaviour with a parameter to getInvoice like allowSlippage would be helpful.

TL;DR:
- add to getInvoice response: amountPaid (string) and isFullyPaid (bool) and a request Parameter to getInvoice: allowSlippage (bool)

Lets discuss this 😊

**Comment by pavlenex** [comment]
Why opening a discussion when there's already an issue for it?

**Reply by mp21-org** [reply]
The issue is on the PHP Greenfield Client and we would like to see this in the btcpayserver API and not in the client. (need for extra PHP bcmath extension if we do it etc.)
